usage-rules.md

```diff
@@ -0,0 +1,534 @@
+mix ash_authentication.add_strategy password
+mix ash_authentication.add_strategy magic_link
+mix ash_authentication.add_strategy password magic_link  # multiple
+
+# Custom user resource
+mix ash_authentication.add_strategy password --user MyApp.Accounts.User
+
+# OAuth2 NOT supported - requires manual setup
+```
+
+**What mix tasks do:**
+- Add required attributes and identities
+- Configure strategy in authentication block
+- Generate actions and sender modules
+- Add dependencies (bcrypt_elixir for password)
+- Run ash.codegen
+
+## Working with Authentication
+
+### Strategy Protocol
+```elixir
+# Get and use strategies
+strategy = AshAuthentication.Info.strategy!(MyApp.User, :password)
+{:ok, user} = AshAuthentication.Strategy.action(strategy, :sign_in, params)
+
+# List strategies
+strategies = AshAuthentication.Info.strategies(MyApp.User)
+```
+
+### Token Operations
+```elixir
+# User/subject conversion
+subject = AshAuthentication.user_to_subject(user)
+{:ok, user} = AshAuthentication.subject_to_user(subject, MyApp.User)
+
+# Token management
+AshAuthentication.TokenResource.revoke(MyApp.Token, token)
+```
+
+### Web Integration
+```elixir
+# Create auth plug
+defmodule MyAppWeb.AuthPlug do
+  use AshAuthentication.Plug, otp_app: :my_app
+end
+
+# Add to router
+forward "/auth", MyAppWeb.AuthPlug
+```
+
+### Policies
+```elixir
+policies do
+  bypass AshAuthentication.Checks.AshAuthenticationInteraction do
+    authorize_if always()
+  end
+end
+```
+
+## Common Implementation Patterns
+
+### Pattern: Multiple Authentication Methods
+When users need multiple ways to authenticate:
+
+```elixir
+authentication do
+  tokens do
+    enabled? true
+    token_resource MyApp.Accounts.Token
+  end
+
+  strategies do
+    password :password do
+      identity_field :email
+      hashed_password_field :hashed_password
+    end
+
+    github do
+      client_id MyApp.Secrets
+      client_secret MyApp.Secrets
+      redirect_uri MyApp.Secrets
+      identity_resource MyApp.Accounts.UserIdentity
+    end
+
+    magic_link do
+      identity_field :email
+      sender MyApp.MagicLinkSender
+    end
+  end
+end
+```
+
+### Pattern: OAuth2 with User Registration
+When new users can register via OAuth2:
+
+```elixir
+actions do
+  create :register_with_github do
+    argument :user_info, :map, allow_nil?: false
+    argument :oauth_tokens, :map, allow_nil?: false
+    upsert? true
+    upsert_identity :email
+
+    change AshAuthentication.GenerateTokenChange
+    change fn changeset, _ctx ->
+      user_info = Ash.Changeset.get_argument(changeset, :user_info)
+
+      changeset
+      |> Ash.Changeset.change_attribute(:email, user_info["email"])
+      |> Ash.Changeset.change_attribute(:name, user_info["name"])
+    end
+  end
+end
+```
+
+### Pattern: Multitenant Authentication
+When using Ash multitenancy:
+
+```elixir
+# Pass tenant in strategy actions
+{:ok, user} = AshAuthentication.Strategy.action(strategy, :sign_in, params, tenant: tenant)
```

Comment:
this manual providing of tenant probably won't be that common. Lets just remove the multi tenant section for now.